cmake/KokkosConfigCommon.cmake.in

```diff
@@ -209,6 +209,290 @@ int main()
+#       DEPENDENT_LIBRARY     --> input: The library to search for in the target properties
+#       SEARCH_DIR            --> input: the root dir to start the recursion from
+#
+function(kokkos_get_directory_targets FOUND_COMPILE_TARGETS FOUND_LINK_TARGETS DEPENDENT_LIBRARY SEARCH_DIR)
+
+  # recursively visit subdirs
+  get_property(subdirs DIRECTORY ${SEARCH_DIR} PROPERTY SUBDIRECTORIES)
+  foreach(subdir ${subdirs})
+    kokkos_get_directory_targets(subdir_compile_targets subdir_link_targets ${DEPENDENT_LIBRARY} ${subdir})
+    list(APPEND compile_targets ${subdir_compile_targets})
+    list(APPEND link_targets ${subdir_link_targets})
+  endforeach()
+
+  #check if target compiles with or links to Kokkos::kokkos
+  get_property(parent_targets DIRECTORY ${SEARCH_DIR} PROPERTY BUILDSYSTEM_TARGETS)
+  foreach(target IN LISTS parent_targets)
+    set(compile_dependency OFF)
+    set(link_dependency OFF)
+    # if a target lists Kokkos::kokkos in the libs it will have to be compiled with our settings no matter private/public/interface
+    get_target_property(dependencies ${target} LINK_LIBRARIES)
+    if(${DEPENDENT_LIBRARY} IN_LIST dependencies)
+      set(compile_dependency ON)
+    endif()
+
+    kokkos_check_dependency(compile_dependency INTERFACE_LINK_LIBRARIES ${DEPENDENT_LIBRARY} ${target})
+    kokkos_check_dependency(link_dependency LINK_LIBRARIES ${DEPENDENT_LIBRARY} ${target})
+    if(compile_dependency)
+      list(APPEND compile_targets ${target})
+    endif()
+    if(link_dependency)
+      list(APPEND link_targets ${target})
+    endif()
+  endforeach()
+
+  # push results out of recursive function
+  set(${FOUND_COMPILE_TARGETS} ${compile_targets} PARENT_SCOPE)
+  set(${FOUND_LINK_TARGETS} ${link_targets} PARENT_SCOPE)
+endfunction()
+
+# this function sets the source and target properties for all targets that link to a given library
+#
+#       PUBLIC          --> input: set downstream sources and targets to have a public dependency on kokkos. If not given a private dependency is assumed
+#       LIBRARY         --> input: name of the library downstream targets need to link to for this function to set the properties.
+#       USE_LAUNCHER    --> input: Command for a launcher to use
+#
+function(kokkos_set_dependent_library_properties)
+  cmake_parse_arguments(INP "PUBLIC" "LIBRARY" "USE_LAUNCHER" ${ARGN})
+
+  if(NOT INP_LIBRARY)
+    message(FATAL_ERROR "'kokkos_set_dependent_library_properties' needs a LIBRARY argument")
+  endif()
+
+  kokkos_get_directory_targets(COMPILE_TARGETS LINK_TARGETS ${INP_LIBRARY} ${PROJECT_SOURCE_DIR})
+
+  if(INP_PUBLIC)
+    if(@Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE@)
+      foreach(target IN LISTS COMPILE_TARGETS)
+        get_target_property(target_sources ${target} SOURCES)
+        get_target_property(target_dir ${target} SOURCE_DIR)
+        list(FILTER target_sources INCLUDE REGEX "\\.cpp$|\\.cxx$")
```

Comment:
`cc` is also pretty common.

Reply:
I am still thinking about letting the user provide this regex and default to what CMake considers to be CXX files